### ISSUE TITLE
csv.lowlevel.headers pipe does not preserve line numbers

### ISSUE THREAD

**Issue by satorg** [satabin/fs2-data#266]
Version: 1.3.0
Scala: 3.1.1-RC

Example:
```scala
import cats.effect._
import cats.syntax.all._
import fs2._
import fs2.io.file._
import fs2.data.csv

object TestApp extends IOApp.Simple {
  override def run: IO[Unit] =
    Files[IO]
      .readAll(Path("some_valid_csv_path"))
      .through(text.utf8.decode)
      .through(csv.lowlevel.rows())
      .evalTap { row =>
        IO(println(s"[1] ${row.line}"))
      }
      .through(csv.lowlevel.headers[IO, String])
      .evalTap { row =>
        IO(println(s"[2] ${row.line}"))
      }
      .compile
      .drain
}
```

Actual output:
```
[1] Some(1)
[1] Some(2)
[2] None
[1] Some(3)
[2] None
[1] Some(4)
[2] None
[1] Some(6)
[2] None
...
```

Expected output:
```
[1] Some(1)
[1] Some(2)
[2] Some(2)
[1] Some(3)
[2] Some(3)
[1] Some(4)
[2] Some(4)
[1] Some(6)
[2] Some(6)
...
```

**Comment by ybasket** [comment]
Thank you for reporting this! I had a brief look and luckily it was very easy to fix, see #267.

**Comment by satabin** [comment]
I just released this fix in version 1.3.1. It should be available soon in sonatype repositories. Thanks for reporting.

**Comment by satorg** [comment]
@satabin yes, it works well now, thank you a lot!
